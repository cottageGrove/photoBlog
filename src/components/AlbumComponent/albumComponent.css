


.box {
    width: 150px;
    height: 150px;
    background-color: #ededed;
    margin-bottom: 20px;

    -moz-transition: all .2s ease-in;
    -o-transition: all .2s ease-in;
    -webkit-transition: all .2s ease-in;
    transition: all .2s ease-in;
}

.box:hover {
    background:#FAFAFA;
    transform: scale(1.02);
    cursor: pointer;
}

.album-title {
    width: 150px;
    height: 78px;
    color: black;
    -moz-transition: all .2s ease-in;
    -o-transition: all .2s ease-in;
    -webkit-transition: all .2s ease-in;
    transition: all .2s ease-in;
}

.album-title:hover {
    color:purple;
    cursor: pointer;

}

@media (max-width: 768px) { 
    .box {
        width: 300px;
        height: 300px;
        background-color: #ededed;
        margin-bottom: 20px;
    }

    .album-title {
        width: 300px;   
    }
}

