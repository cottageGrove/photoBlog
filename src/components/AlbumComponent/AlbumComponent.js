import React, {Component} from 'react'
import {Link} from 'react-router-dom'

import PhotosPageComponent from '../../page-components/PhotosPageComponent/PhotosPageComponent'
import './albumComponent.css'

import {
    BrowserRouter as Router, 
    Switch, 
    Route,
  } from 'react-router-dom'


class AlbumComponent extends Component {

    constructor(props) {
        super(props)
        this.state = {
            userId: props.userId,
            albumId: props.id,
            title: props.title
        }

        this.goToPhotos = this.goToPhotos.bind(this)
    }

    goToPhotos() {
        //Connect to redux
    }

    render() {

        return (
            <div onClick={this.goToPhotos}>
                <Link to={{pathname: `/photon/albums/${this.props.albumId}/photos`}}>
                    <div className="box">

                    </div>

                    <div className ="album-title">
                        {this.props.title}
                    </div>
                                        
                </Link>
            </div>

        )
    }
}

export default AlbumComponent