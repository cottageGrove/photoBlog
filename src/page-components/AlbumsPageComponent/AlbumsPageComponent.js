import React, {Component} from 'react'
import AlbumComponent from '../../components/AlbumComponent/AlbumComponent'
import PhotosPageComponent from '../PhotosPageComponent/PhotosPageComponent'
import { withRouter } from 'react-router'
import '../../page-components/pageComponent.css'

import {
    BrowserRouter as Router, 
    Switch, 
    Route,
} from 'react-router-dom'

import {
    REQUEST_ALBUMS,
    RECEIVE_ALBUMS,
    fetchAlbumsIfNeeded
} from '../../redux/actions'
import { connect } from 'react-redux'


class AlbumsPageComponent extends Component {

    constructor(props) {
        super(props)
        this.state = {
            userId : props.userId,
            albums: []
        }

        this.getAlbums = this.getAlbums.bind(this)
    }

    componentDidMount() {
        this.getAlbums()
    }

    async getAlbums() {

        const {dispatch} = this.props
        //need to assign the UserId depending on the input 

        dispatch(fetchAlbumsIfNeeded(1))

        const {albums} = this.props
        this.setState({albums: albums})
    }

    render() {

        const albums = this.state.albums

        console.log(albums)

        const albumCollection = albums.map((album) => {
            return <AlbumComponent userId={album.userId} albumId={album.id} key={album.id} title={album.title}></AlbumComponent>
        })

        
        return (


            <div>
                 <div>
                    <div className="page-title">
                        <div>
                            <h2>
                                Album
                            </h2>
                        </div>
                    </div>

                    <div className="grid-container">

                        {albumCollection}
                    </div>
                </div>
            </div>
           
    
        )
    }
}




export default AlbumsPageComponent