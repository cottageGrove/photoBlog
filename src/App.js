import React from 'react';
import logo from './logo.svg';
import './App.css';

import {
  BrowserRouter, 
  Switch, 
  Route,
} from 'react-router-dom'


import AlbumsPageComponent from './page-components/AlbumsPageComponent/AlbumsPageComponent';
import PhotosPageComponent from './page-components/PhotosPageComponent/PhotosPageComponent';
import PostsPageComponent from './page-components/PostsPageComponent/PostsPageComponent'

import NavigationBarComponent from './components/NavigationBarComponent/NavigationBarComponent'
import AlbumComponent from './components/AlbumComponent/AlbumComponent';
import VisibleAlbumsPageComponent from './page-components/VisibleAlbumsPageComponent';


const userId = 1

function App() {
  return (

    <BrowserRouter>

      <NavigationBarComponent></NavigationBarComponent>


      <Switch>
        <Route path='/photon/albums/:albumId/photos'>
          <PhotosPageComponent></PhotosPageComponent>
        </Route>
        <Route path='/photon/albums'>
          <VisibleAlbumsPageComponent></VisibleAlbumsPageComponent>

          {/* <AlbumsPageComponent userId={userId}></AlbumsPageComponent> */}
        </Route>


        <Route path='/photon/posts/:postId/comments'>

        </Route>

        <Route path='/photon/posts'> 
          <PostsPageComponent> </PostsPageComponent>
        </Route>

        
      </Switch>
    </BrowserRouter>
 

  );
}

export default App;
