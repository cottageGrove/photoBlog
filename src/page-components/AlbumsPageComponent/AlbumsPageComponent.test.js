import React from 'react'
import renderer from "react-test-renderer"
import { shallow, mount } from 'enzyme'
// import {expect} from 'chai'
import  AlbumsPageComponent  from './AlbumsPageComponent'
import { Route, Link, Router} from "react-router-dom";
import { createMemoryHistory } from 'history'


import { render, fireEvent } from '@testing-library/react'
import configureMockStore from "redux-mock-store";
import { Provider } from "react-redux";


import App from '../../App'
import {act} from 'react-dom/test-utils'

import {MemoryRouter} from 'react-router-dom'


import AlbumComponent from '../../components/AlbumComponent/AlbumComponent'


const mockStore = configureMockStore();
const store = mockStore({});

describe('<AlbumPageComponent>', () => {
    describe('Albums', () => {
        it('whether Album Component is in AlbumsPageComponent', () => {
            const wrapper = mount(<AlbumsPageComponent />)


            expect(wrapper.find('AlbumComponent')).toBeTruthy()

            // expect(wrapper).toContainMatchingElement('AlbumComponent')

        })
    })
})


test('navigate to /photos', () => {
    const history = createMemoryHistory()
    const { container, getByText } = render(
      <Provider store={store}>
        <Router history={history}>
            <App />
        </Router>
      </Provider>
    )
    // verify page content for expected route
    // often you'd use a data-testid or role query, but this is also possible
    expect(container.innerHTML).toMatch('Albums')
  
    fireEvent.click(getByText(/Albums/))
  
    // check that the content changed to the new page
    expect(container.innerHTML).toMatch('Photos')
  })

// test('AlbumComponent', () => {

//         let history, location;
//         const albumWrapper = shallow( 
 
//             <MemoryRouter initialEntries={["/my/initial/route"]}>
//                 <App />
//                 <Route
//                 path="*"
//                 render={({ history, location }) => {
//                     history = history;
//                     location = location;
//                     return null;
//                 }}
//                 />
//             </MemoryRouter>)


//         act(()=> {

//             // const link = albumWrapper.find('Link')
//             // link.simulate('click')

//             //need to assert some changing value

//         })

//         expect(location.pathname).toBe('/photos')
//         const searchParams = new URLSearchParams(location.search)
//         expect(searchParams.has('userId')).toBe(true)



// })
