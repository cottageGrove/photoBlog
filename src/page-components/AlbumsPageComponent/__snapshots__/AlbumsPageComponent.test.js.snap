// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlbumPageComponent contains albums 1`] = `
<div>
  <div>
    <div
      className="page-title"
    >
      <div>
        <h2>
          Album
        </h2>
      </div>
    </div>
    <div
      className="grid-container"
    />
  </div>
</div>
`;
